server/utils/bot/flower.go

```diff
@@ -0,0 +1,105 @@
+package bot
+
+import "github.com/mattermost/mattermost-plugin-mscalendar/server/utils/flow"
+
+type Flower interface {
```

Comment:
Since a "flower" controls the start, cancel, and follow the next step of all the flows, I think it is better to call it controller. And since the interface is outside the flow package, I decided to call it FlowController.

Nevertheless, I can move the interface to the flow package, and leave in the bot package only the implementation. That way the interface would be just "controller", and the file would keep being flow_controller.go